packages/headless/src/controllers/insight/smart-snippet/headless-insight-smart-snippet-interactive-inline-links.ts

```diff
@@ -0,0 +1,144 @@
+import {InsightEngine} from '../../../app/insight-engine/insight-engine';
+import {questionAnswering, search} from '../../../app/reducers';
+import {logOpenSmartSnippetSuggestionInlineLink} from '../../../features/question-answering/question-answering-insight-analytics-actions';
+import {
+  QuestionAnsweringSection,
+  SearchSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {getObjectHash} from '../../../utils/utils';
+import {
+  buildInteractiveResultCore,
+  InteractiveResultCore,
+} from '../../core/interactive-result/headless-core-interactive-result';
+import {InlineLink} from '../../core/smart-snippet-questions-list/headless-core-smart-snippet-questions-list';
+
+/**
+ * @internal
+ */
+export interface SmartSnippetInteractiveInlineLinksOptions {
+  selectionDelay?: number;
+}
+
+/**
+ * @internal
+ */
+export interface SmartSnippetInteractiveInlineLinksProps {
+  options?: SmartSnippetInteractiveInlineLinksOptions;
+}
+
+/**
+ * @internal
+ */
+export interface SmartSnippetInteractiveInlineLinks {
+  selectInlineLink(link: InlineLink, questionAnswerId?: string): void;
+  beginDelayedSelectInlineLink(
+    link: InlineLink,
+    questionAnswerId?: string
+  ): void;
+  cancelPendingSelectInlineLink(
+    link: InlineLink,
+    questionAnswerId?: string
+  ): void;
+}
+
+/**
+ * @internal
+ */
+export function buildSmartSnippetInteractiveInlineLinks(
+  engine: InsightEngine,
+  props?: SmartSnippetInteractiveInlineLinksProps
+): SmartSnippetInteractiveInlineLinks {
+  if (!loadSmartSnippetInteractiveInlineLinksReducer(engine)) {
+    throw loadReducerError;
+  }
+
+  const getState = () => engine.state;
+
+  const clickedRelatedQuestions = new Set<string>();
+  const inlineLinkWasClicked = (linkId: string) => {
+    if (clickedRelatedQuestions.has(linkId)) {
+      return true;
+    }
+    clickedRelatedQuestions.add(linkId);
+    return false;
+  };
+
+  let lastSearchResponseId: string | null = null;
+  const resetInteractiveResultsIfSearchResponseChanged = (
+    currentSearchResponseId: string
+  ) => {
+    if (lastSearchResponseId !== currentSearchResponseId) {
+      lastSearchResponseId = currentSearchResponseId;
+      interactiveResultsPerInlineLink = {};
+      clickedRelatedQuestions.clear();
+    }
+  };
+
+  const buildInlineLinkInteractiveResult = (
+    link: InlineLink,
+    linkId: string,
+    questionAnswerId?: string
+  ) =>
+    buildInteractiveResultCore(
+      engine,
+      {options: {selectionDelay: props?.options?.selectionDelay}},
+      () => {
+        if (inlineLinkWasClicked(linkId)) {
+          return;
+        }
+        // TODO: Replace with the comment below after creating the logOpenSmartSnippetInlineLink action for the insight use case.
```

Comment:
Add a Jira in the TODO comment if you won't complete it in this PR

Reply:
I thought I had fixed that in my prior PR???

Reply:
yep this has been addressed.
this an outdated piece of code.